x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/validate_rule_actions_field/validate_rule_actions_field.ts

```diff
@@ -51,3 +54,38 @@ export const validateRuleActionsField =
       };
     }
   };
+
+/**
+ * Throttles validation by canceling previous validation requests. Necessary
+ * to prevent error validation flashing when first adding an action until root
+ * cause of https://github.com/elastic/kibana/issues/142217 is found
+ *
+ * See docs for details:
+ * https://docs.elastic.dev/form-lib/examples/validation#cancel-asynchronous-validation
+ *
+ * @param actionTypeRegistry
+ */
+export const throttledValidateRuleActionsField =
+  (actionTypeRegistry: ActionTypeRegistryContract) =>
+  (
+    data: ValidationFuncArg<ActionsStepRule | RuleActionsFormData>
+  ): Promise<ValidationError | void | undefined> => {
+    let isCanceled = false;
+    const promise: Promise<ValidationError | void | undefined> & { cancel?(): void } = new Promise(
+      (resolve) => {
+        setTimeout(() => {
+          if (isCanceled) {
+            resolve();
+          } else {
+            resolve(validateRuleActionsField(actionTypeRegistry)(data));
+          }
+        }, 100);
```

Comment:
nit: the best option is providing timeout as function parameter and define it as a constant like `DEFAULT_VALIDATION_TIMEOUT`.